edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -2335,7 +2335,7 @@ public String getAssignmentScoreStringByNameOrId(final String gradebookUid, fina
 			return null;
 		}
 
-		if (score == null) {
+		if (!isAssignmentDefined(gradebookUid, assignmentName) && score == null) {
```

Comment:
I would switch the order of the conditions here since (score = null) is cheaper than isAssignmentDefined

Reply:
Done!